scripts/contractInteraction/staking_vesting.py

```diff
@@ -366,11 +367,32 @@ def updateLockedSOV():
     # sendWithMultisig(conf.contracts['multisig'], lockedSOV.address, data, conf.acct)
 
 #receiver is usually the multisig
-def governanceWithdrawVesting( vesting,  receiver):
+def governanceDirectWithdrawVesting( vesting,  receiver):
     stakingProxy = Contract.from_abi("Staking", address=conf.contracts['Staking'], abi=Staking.abi, owner=conf.acct)
-    data = stakingProxy.governanceWithdrawVesting.encode_input( vesting,  receiver)
-    print(data)
-    sendWithMultisig(conf.contracts['multisig'], conf.contracts['Staking'], data, conf.acct)
+    vesting = Contract.from_abi("VestingLogic", address=vesting, abi=VestingLogic.abi, owner=conf.acct)
+
+    DAY = 24 * 60 * 60
+    TWO_WEEKS = 2 * 7 * DAY
+
+    vestingStartDate = vesting.startDate()
+    vestingEnd = vesting.endDate()
+    vestingCliff = vesting.cliff()
+    defaultStart = vestingStartDate + vestingCliff
+
+    maxIterations = stakingProxy.getMaxVestingWithdrawIterations()
+    maxIterationsEnd = defaultStart + (TWO_WEEKS * maxIterations)
+
+    counter = 1
+
+    if vestingEnd > maxIterationsEnd:
+        counter = math.ceil(maxIterationsEnd / vestingEnd)
+
+    for x in range(counter):
+        startFrom = defaultStart
+        data = stakingProxy.governanceDirectWithdrawVesting.encode_input( vesting,  receiver, startFrom)
+        print(data)
+        sendWithMultisig(conf.contracts['multisig'], conf.contracts['Staking'], data, conf.acct)
+        startFrom = startFrom + (maxIterations * TWO_WEEKS)
```

Comment:
it won't work if counter > 1 as `startFrom` is reassigned to the `defaultStart` in the beginning of the cycle

Reply:
updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/7f1c269ee40b1d2f3f864b07c574360839a9d270)